core/src/main/java/org/jruby/util/JRubyClassLoader.java

```diff
@@ -86,6 +90,8 @@ public void addURL(URL url) {
                 out.close();
                 in.close();
                 url = f.toURI().toURL();
+
+                cachedJarPaths.add(url.getPath());
```

Comment:
not sure how much its relevant but URL had issues - hopefully not for file: protocol but it might be worth a check
... why `org.jruby.util.URLUtil.getPath` exists and whether it should be used here

Reply:
I looked into the issue, and from what I was able to understand, people recommend using [Paths](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Paths.html)/Path objects if you're on java 7 or higher. Does jruby need to support java 6?

I also saw recommendations to use `url.toURI().getSchemeSpecificPart()` to get the String path, which is what `URLUtil.getPath` does, so it might be easier, and safe, to just use that utility function here

Reply:
OK, feel free to update to using `URLUtil` than. also JRuby 9 only supported Java >= 7

Reply:
Updated